addons/product/views/product_attribute_views.xml

```diff
@@ -24,9 +24,9 @@
                     <button class="oe_stat_button" name="action_open_related_products"
                             type="object" icon="fa-bars"
                             invisible="not number_related_products">
-                        <div class="o_stat_info">
-                            <span class="o_stat_value"><field name="number_related_products"/></span>
-                            <span class="o_stat_text">Related Products</span>
+                        <div>
+                            <span class="me-1"><field name="number_related_products"/></span>
+                            <span>Products</span>
```

Comment:
Why you need to remove these classes? :thinking:

Reply:
Since we needed both the span's on single line I had to remove classes "o_stat_info", "o_stat_value", "o_stat_text" . These classes were adding property display:"block" to those spans due to which they were in separate lines. However i checked stat buttons in other modules and they all have both the text and number in different lines.